## 86% of end-end tests have passed
```
Total executed tests: 7
Failed tests: 1
Expected to fail tests: 0
Passed tests: 6
```
```
IDs of failed tests: 702843 
```
<h3>Failed tests (1)</h3><details><summary>Click to expand</summary><br/><li><a href="https://ci.status.im/job/status-mobile/job/e2e/job/status-app-prs-rerun/parambuild/?BRANCH_NAME=fix/currency-fetch-delay&PR_ID=21239&APK_NAME=21239.apk&TR_CASE_IDS=702843">Rerun failed tests</a></li><br/><h4>Class TestCommunityMultipleDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3668946">test_community_message_edit</a>, id: 702843 </b></td></tr><tr><td><p><blockquote><div>Device 2: Tap on found: Button </div><div>Device 2: Attempt 0 is successful clicking //*[contains(@text, 'user_1')]/ancestor::*[@content-desc='activity']/*[@content-desc="accept-contact-request"] </div></blockquote></p>

```
Test setup failed: critical/chats/test_public_chat_browsing.py:320: in prepare_devices
    self.home_2.handle_contact_request(self.username_1)
../views/home_view.py:422: in handle_contact_request
    self.close_activity_centre.wait_for_rendering_ended_and_click()
../views/base_element.py:163: in wait_for_rendering_ended_and_click
    self.wait_for_visibility_of_element(20)
../views/base_element.py:145: in wait_for_visibility_of_element
    .until(expected_conditions.visibility_of_element_located((self.by, self.locator)))
../../../../status-app-prs@tmp/venv/lib/python3.10/site-packages/selenium/webdriver/support/wait.py:86: in until
    value = method(self._driver)
../../../../status-app-prs@tmp/venv/lib/python3.10/site-packages/selenium/webdriver/support/expected_conditions.py:152: in _predicate
    return _element_if_visible(driver.find_element(*locator))
../../../../status-app-prs@tmp/venv/lib/python3.10/site-packages/selenium/webdriver/support/expected_conditions.py:175: in _element_if_visible
    return element if element.is_displayed() == visibility else False
../../../../status-app-prs@tmp/venv/lib/python3.10/site-packages/appium/webdriver/webelement.py:81: in is_displayed
    return self._execute(Command.IS_ELEMENT_DISPLAYED)['value']
../../../../status-app-prs@tmp/venv/lib/python3.10/site-packages/selenium/webdriver/remote/webelement.py:395: in _execute
    return self._parent.execute(command, params)
../../../../status-app-prs@tmp/venv/lib/python3.10/site-packages/selenium/webdriver/remote/webdriver.py:345: in execute
    self.error_handler.check_response(response)
../../../../status-app-prs@tmp/venv/lib/python3.10/site-packages/appium/webdriver/errorhandler.py:122: in check_response
    raise exception_class(msg=message, stacktrace=format_stacktrace(stacktrace))
 An unknown server-side error occurred while processing the command. Original error: Could not proxy command to the remote server. Original error: socket hang up
E   Stacktrace:
E   UnknownError: An unknown server-side error occurred while processing the command. Original error: Could not proxy command to the remote server. Original error: socket hang up
E       at UIA2Proxy.command (/mnt/sauce/appium/appium-v2.0.0/packages/base-driver/lib/jsonwp-proxy/proxy.js:355:13)
E       at processTicksAndRejections (node:internal/process/task_queues:95:5)
E       at AndroidUiautomator2Driver.commands.getAttribute (/mnt/sauce/appium/appium-v2.0.0/appium_home/linux/node_modules/appium-uiautomator2-driver/lib/commands/element.js:17:10)
E       at AndroidUiautomator2Driver.commands.elementDisplayed (/mnt/sauce/appium/appium-v2.0.0/appium_home/linux/node_modules/appium-uiautomator2-driver/lib/commands/element.js:21:17)
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/62681860785844bbad784e15c099f05a?auth=c93c354f5aa21891f65380487a3d4bf6">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/fbbee8108761491bb0151d5ba6e547c1?auth=8a5a4bbc08ea7b622c0d82fc10d0b7ee">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table></details><h3>Passed tests (6)</h3><details><summary>Click to expand</summary><br/><br/><h4>Class TestWalletMultipleDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3668949">test_wallet_send_asset_from_drawer</a>, id: 727230 </b></td></tr><tr><td></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3668948">test_wallet_send_eth</a>, id: 727229 </b></td></tr><tr><td></td></tr></tbody></table><h4>Class TestWalletOneDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3668950">test_wallet_add_remove_regular_account</a>, id: 727231 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/91d4fe661ead4f8caf8c67bf4865483c?auth=d2bb2f5309321cdedab4ec26eac03094#39">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestOneToOneChatMultipleSharedDevicesNewUi:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3668945">test_1_1_chat_non_latin_messages_stack_update_profile_photo</a>, id: 702745 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/e8b81e97a01c457394e50856872fc5f8?auth=f3bf7f9dd68c38f88c569684a0b4af7b#105">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/48c9b3ecff3c433abd26f908029b6894?auth=7a5abc158798e47344115f96a2055d83#92">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestCommunityOneDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3668947">test_restore_multiaccount_with_waku_backup_remove_switch</a>, id: 703133 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/d27e8bfb2dbe4ae9aeaa941ef15519b9?auth=cb1715eecf8dab66e9a092c0f0ef77d0#83">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3668944">test_community_copy_and_paste_message_in_chat_input</a>, id: 702742 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/d27e8bfb2dbe4ae9aeaa941ef15519b9?auth=cb1715eecf8dab66e9a092c0f0ef77d0#54">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table></details>

--- a/src/status_im/contexts/profile/login/events.cljs
+++ b/src/status_im/contexts/profile/login/events.cljs
@@ -62,6 +62,9 @@
                  ;; loading chats and communities. This globally helps alleviate
                  ;; stuttering immediately after login.
                  [:dispatch-later [{:ms 500 :dispatch [:wallet/initialize]}]]
+                 ;; Fetching the currencies along with wallet initialization as
+                 ;; we rely on it for displaying currency symbol
+                 [:dispatch-later [{:ms 500 :dispatch [:settings/get-currencies]}]]
 
                  [:logs/set-level log-level]
 
@@ -95,7 +98,6 @@
            [:dispatch-later [{:ms 1500 :dispatch [:profile.login/non-critical-initialization]}]]
            [:dispatch [:network/check-expensive-connection]]
            [:profile.settings/get-profile-picture key-uid]
-           [:settings/get-currencies]
            (when (ff/enabled? ::ff/wallet.wallet-connect)
              [:dispatch [:wallet-connect/init]])
            (when (ff/enabled? ::ff/wallet.swap)

--- a/src/status_im/contexts/settings/language_and_currency/events.cljs
+++ b/src/status_im/contexts/settings/language_and_currency/events.cljs
@@ -1,6 +1,5 @@
 (ns status-im.contexts.settings.language-and-currency.events
-  (:require [status-im.common.json-rpc.events :as json-rpc]
-            [status-im.contexts.settings.language-and-currency.data-store :as data-store]
+  (:require [status-im.contexts.settings.language-and-currency.data-store :as data-store]
             [utils.collection]
             [utils.re-frame :as rf]))
 
@@ -11,8 +10,9 @@
                  :currencies
                  (utils.collection/index-by :id all-currencies))})))
 
-(rf/reg-fx :settings/get-currencies
+(rf/reg-event-fx :settings/get-currencies
  (fn []
-   (json-rpc/call {:method     "appgeneral_getCurrencies"
-                   :on-success [:settings/get-currencies-success]
-                   :on-error   [:log-rpc-error {:event :settings/get-currencies}]})))
+   {:fx [[:json-rpc/call
+          [{:method     "appgeneral_getCurrencies"
+            :on-success [:settings/get-currencies-success]
+            :on-error   [:log-rpc-error {:event :settings/get-currencies}]}]]]}))

